ts4.1/index.d.ts

```diff
@@ -0,0 +1,265 @@
+	 * Dispatches to PascalCase or CamelCase.
+	 *
+	 * @param S a union of string literal types to camelcase.
+	 * @param UsePascalCase if true, the first word is also capitalized.
+	 */
+	type Camelize<S extends string, UsePascalCase extends boolean = false> =
+		UsePascalCase extends true ? PascalCase<S> : CamelCase<S>;
+
+	/**
+	 * @internal
+	 * Digs a set of paths by one depth. Used for processing `stopPaths`.
+	 *
+	 * @param Path a union of string literal types, each of which is a "path" from the root.
+	 * @param S a prefix word. A string literal type or `string`.
+	 *
+	 * @example
+	 * ```
+	 * type T1 = FilterPaths<"foo.bar" | "foo.baz" | "foo.bar.baz" | "other.path", "foo">;
+	 * // => "bar" | "baz" | "bar.baz"
+	 *
+	 * type T2 = FilterPaths<string, "foo">;
+	 * // => string
+	 *
+	 * type T3 = FilterPaths<"foo.bar" | "foo.baz" | "foo.bar.baz" | "other.path", string>;
+	 * // => string
+	 *
+	 * type T3 = FilterPaths<"foo" | "bar", string>;
+	 * // => never
+	 * ```
+	 */
+	type FilterPaths<Path extends string, S extends string> =
+		// We treat `string` as "indeterminate set of strings", so return `string` in this case.
+		string extends Path ? string :
+		// If `S` is indeterminate, we are not sure about the resulting set,
+		// except when there is no path containing `.`.
+		string extends S ?
+		Path extends `${string}.${string}` ? string : never :
+		// Otherwise, do a normal filtering.
+		// If a path starts with `${S}.`, then return the suffix.
+		// Otherwise return never (which means the candidate is rejected)
+		Path extends `${S}.${infer Suffix}` ? Suffix : never;
+
+	/**
+	 * @internal
+	 * Helper type to conditionally camelize a string.
+	 *
+	 * @param S a union of string literal types to camelcase.
+	 * @param PascalCase if true, the first word is also capitalized.
+	 * @param Exclude if it's a union of string literals, the keys matching `Exclude` are kept as-is.
+	 *        If it's unclear which keys to keep, specify `string`.
+	 *
+	 * @example
+	 * ```
+	 * type T1 = CamelcaseUnless<"foo_bar" | "bar_baz", false, never>;
+	 * // => "fooBar" | "barBaz"
+	 *
+	 * type T2 = CamelcaseUnless<"foo_bar" | "bar_baz", false, "bar_baz">;
+	 * // => "fooBar" | "bar_baz"
+	 *
+	 * type T1 = CamelcaseUnless<"foo_bar" | "bar_baz", false, string>;
+	 * // => "foo_bar" | "fooBar" | "bar_baz" | "barBaz"
+	 * ```
+	 */
+	type CamelCaseUnless<S extends string, PascalCase extends boolean, Exclude extends string> =
+		// We treat `string` as "indeterminate set of strings", so return both cases in this case.
+		string extends Exclude ? S | Camelize<S, PascalCase> :
+		S extends Exclude ? S : Camelize<S, PascalCase>;
+
+	/**
+	 * @internal
+	 * Compute return value of camelcaseKeys.
+	 *
+	 * @param T an object or array to camelcase its keys
+	 * @param PascalCase capitalize the first word
+	 * @param Deep recursively camelcase
+	 * @param Exclude don't camelcase these keys
+	 * @param StopPaths don't recursive into these paths
```

Comment:
Follow the correct `@param` syntax. https://github.com/sindresorhus/typescript-definition-style-guide